ui/ducks/send/send.js

```diff
@@ -354,7 +354,7 @@ export const computeEstimatedGasLimit = createAsyncThunk(
     const { send, metamask } = state;
     const isNonStandardEthChain = getIsNonStandardEthChain(state);
     const chainId = getCurrentChainId(state);
-    if (send.stage !== SEND_STAGES.EDIT) {
+    if (send.stage) {
```

Comment:
Can I get short explanation why this change was required?

Reply:
@david0xd This change was required because when the user wants to modify the original send amount to something different, `gasLimit` will not reset to default 21000, because when the user click the `< Edit` button at the top left, `send.stage` will be in status `EDIT` and than, when the user is modifying the original send amount to something different, function `estimateGasLimitForSend` will be called and recompute the `gasLimit` estimate to be as accurate as possible.

Reply:
I think there are at least two cases when `send.stage === SEND_STAGES.EDIT` that we don't want new estimation to happen. First, is if the gas limit was explicitly proposed by a dapp. The second is if the user has already explicitly set the gas limit.

For the first case, we can check the transaction data for a `dappSuggestedGasFees.gas` property.

For the second, I think we might have to modify the `onSubmit` function in `ui/components/app/edit-gas-popover/edit-gas-popover.component.js` so that if the gas limit was changed by the user, then we record a "userEditedGasLimit" on the txMeta, so that it can be accessed when editing the transaction.

Both of these will only work if the transaction metadata is available when editing the transaction.

Reply:
@danjm Thank you Dan for commenting this issue. My understanding of these cases are:

For the first case, I checked the transaction data for a `dappSuggestedGasFees.gas` property in the code. I run this case several times and I think it doesn't estimate new gas limit and gas limit is the same as `dapp` is explicitly proposed.

For the second, let's say gas limit for a transaction is 21000, the user explicitly sets gas limit to 30,000 and click `Save`, should that value of 30,000 remain in the gas limit field if he click `Edit` again where the gas limit field is located? If this is okay and if I understand correctly this second case, it works as in the above exposition and we don't need to modify the `onSubmit` function in `ui/components/app/edit-gas-popover/edit-gas-popover.component.js`. If there is another solution for this second case, please let me know.

Reply:
@VSaric The issue I describe can be seen in the two videos below (which I recorded on your branch):


https://user-images.githubusercontent.com/7499938/145382988-a949ef4d-5ef1-494b-b3b0-a2763ef7e1a7.mp4


https://user-images.githubusercontent.com/7499938/145382991-126b7eef-47a6-40ee-a45e-275cd799ae99.mp4

Reply:
To recreate:

1) Create a token from the test dapp
2) Create a transaction to transfer the token
3) Edit the gas limit
4) Click 'Edit' in the top left to go to the send/edit screen
5) change the transfer amount to `0` or from `0`
6) Go back to the confirm screen, open the confirm popup, and see that the gas limit has changed from what it was edited to before

Reply:
The reason we want to prevent changing the gas limits in these cases is this is that currently, on the send/edit screen, we don't show the user any information about the gas limit. So if the user manually edited the gas limit, we don't want to change it without making it clear to them that we are doing so. We also don't want to change a dapp suggested gas limit, in case a dapp has intentionally set the gas limit very high because they know that estimations will be inaccurate.

Reply:
So I was thinking that the `if (send.stage) {` condition needs to change to something like `if (send.stage !== SEND_STAGES.EDIT || !dappSuggestedGasFees?.gas || !userEditedGasLimit) {`

Reply:
Thank you @danjm for added comments here to further explanation of this cases. 
I made changes related to this explanation.
You can see changes here: e2afb797e92694ced2d0a1565681202efd7acd52
Please take a look and review and let me know what do you think about this.